app/js/lib/foxbox/recipes.js

```diff
@@ -0,0 +1,300 @@
+'use strict';
+
+// @todo Proxy recipes in the db and emit events on change (see services).
+
+// Private members.
+const p = {
+  // Private properties.
+  settings: Symbol('settings'),
+  net: Symbol('net')
+};
+
+export default class Recipe {
+  constructor(props) {
+    this[p.settings] = props.settings;
+    this[p.net] = props.net;
+  }
+
+  /**
+   * Returns a promise resolving to a list of recipes.
+   *
+   * @return {Promise}
+   */
+  getAll() {
+    return this[p.net].fetchJSON(
+      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/services`)
+      // Get all recipes.
+      .then(services => services.filter(service =>
+        service.adapter === 'thinkerbell-adapter' &&
+        service.id !== 'thinkerbell-root-service'
+      ))
+      // Fetch their respective enabled status.
+      .then(recipes => {
+        const promises = recipes.map(recipe => {
+          const payload = [
+            { id: `${recipe.id}/get_enabled` }
+          ];
+
+          return this[p.net].fetchJSON(
+            `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/` +
+            'channels/get',
+            'PUT',
+            payload);
+        });
+
+        return Promise.all(promises)
+          .then(servicesEnabled => {
+            return servicesEnabled;
+          })
+          // Map all the recipes to a more user-friendly format.
+          .then(servicesEnabled => recipes.map((recipe, id) => ({
+            id: recipe.id,
+            label: recipe.id,
+            enabled: servicesEnabled[id][`${recipe.id}/get_enabled`]
+              .OnOff === 'On'
+          })));
+      });
+  }
+
+  getServicesWithGetters() {
+    return this[p.net].fetchJSON(
+      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/services`)
+      // Get all services but the recipes.
+      .then(services => services.filter(service =>
+        service.adapter != 'thinkerbell-adapter' &&
+        service.adapter != 'webpush@link.mozilla.org'
+      ))
+      // Keep only the services with getters.
+      .then(services => services.filter(service =>
+        Object.keys(service.getters).length
+      ))
+      // User friendly name.
+      .then(services => services.map(service => {
+          switch (service.id) {
+            case 'service:clock@link.mozilla.org':
+              service.name = 'Everyday';
+              break;
+
+            default:
+              service.name = service.id;
+              break;
+          }
+
+          return service;
+        }
+      ))
+      // User friendly getters.
+      .then(services => services.map(service => {
+          switch (service.id) {
+            case 'service:clock@link.mozilla.org':
+              service.getters = [
+                {
+                  label: 'in the morning',
```

Comment:
I think all these user friendly labels should be done in the view itself, so that it can be easily localizable and to not pollute main foxbox lib, but I think we can leave it as is for this PR and improve in the follow ups.